framework/db/src/main/java/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -650,7 +655,7 @@ protected void setField(Object entity, Field field, ResultSet rs, int index) thr
             } else if (type == byte[].class) {
                 field.set(entity, rs.getBytes(index));
             } else {
-                field.set(entity, rs.getObject(index));
+                field.set(entity, _conversionSupport.convertToEntityAttribute(field, rs.getObject(index)));
```

Comment:
is this unconditionally? below a check for the annotation is done (see line 1603)